### PR TITLE
Pysam upgrade and nonREFnonALT fix

### PR DESCRIPTION
**Pull request by NagaComBio**
- Updating the virtual environment path
- Upgrading the pysam version to be compatible with hg38 version
- Adding a nonREFnonALT check to avoid overcount in the BoolCounter class

### CODE REVIEW

File: resources/analysisTools/snvPipeline/filter_PEoverlap.py [hunk 1]
```diff
@@ -96,18 +96,23 @@ def decreaseDP4(remove_base, remove_is_reverse, REF, ALT, DP4rf, DP4rr, DP4af, D
...
-            self.ref += 1
-        else:
-            self.alt += 1
+    def update(self, current_base):
+        if self.REF == current_base:
+            self.ref_count += 1
+        elif self.ALT == current_base:
+            self.alt_count += 1
+
+    def update_nonREFnonALT(self, remove_count):
```
Comment (vinjana) [inline resources/analysisTools/snvPipeline/filter_PEoverlap.py:111]: Why remove_count? Why not simply count?

File: resources/analysisTools/snvPipeline/filter_PEoverlap.py [hunk 2]
```diff
@@ -127,6 +132,7 @@ def performAnalysis(args):
     reference_file = pysam.Fastafile(args.refFileName)
 
     mode = "r"
+    multiple_iterators = False
     if args.alignmentFile[-4:] == ".bam":
```
Comment (vinjana) [inline resources/analysisTools/snvPipeline/filter_PEoverlap.py:136]: I am wondering about this code. What is 4 and 5? Is there some hidden assumption about the format of the filename? A comment would enormeously improve the readability, here.

File: resources/analysisTools/snvPipeline/filter_PEoverlap.py [hunk 3]
```diff
@@ -96,18 +96,23 @@ def decreaseDP4(remove_base, remove_is_reverse, REF, ALT, DP4rf, DP4rr, DP4af, D
 class BoolCounter:
     """ A class for counter objects """ 
 
-    def __init__(self):
-        self.ref = 0
-        self.alt = 0
+    def __init__(self, ref, alt):
```
Comment (vinjana) [inline resources/analysisTools/snvPipeline/filter_PEoverlap.py:99]: What are ref and alt? Are these strings/chars representing bases? If so you could name them `refBase`, which would make this **much** clearer. Capitalized variables are for constants, or not?

**Review by vinjana — APPROVED** [review]
Looks good to me, but I have to admit, I don't know the underlying biology. I guess this is all to cope with some counts that are/should not be counted into the SNV statistics, right? Consequently, I have only superficial comments.

File: resources/analysisTools/snvPipeline/filter_PEoverlap.py (hunk 2 above)
Comment (NagaComBio) [reply resources/analysisTools/snvPipeline/filter_PEoverlap.py:136]: Updated the file extension detection here.

**Comment by NagaComBio** [comment]
Thanks, @vinjana. This is a minor technical update to the counter and updating the env path. Would merge and tag it with 2.1.0

**Pull request merged**